### PR TITLE
Speed up Github Actions

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Currently, our Github Action checks run a lot of redundant commands that take a very long time.

# Solution

- Replace `prepare` command with `prepublishOnly` to run all builds
- Add `prebuild` command to clean the dist state and load pre-commit hooks
- Added a pre-commit hook to test for eslint, styles and format issues before committing
- Added step in our CI build to run `npm run check`

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -56,6 +56,10 @@ jobs:
           issue_labels: vulnerability
           dedupe_issues: true
 
+      - name: npm run check
```
Comment (kopijunkie) [inline .github/workflows/CI.yml:59]: A bit confusing to have the job's name same as the command it's running IMO. 
```suggestion
      - name: Run linters
```

File: package.json [hunk 2]
```diff
@@ -12,14 +12,15 @@
...
+    "prebuild": "npm run clean",
     "build": "webpack --quiet --display-error-details --bail",
     "build:auth": "SDK_ENV=Auth webpack --quiet --display-error-details --bail",
     "build:dev": "NODE_ENV=development npm run build",
     "build:test": "NODE_ENV=test npm run build && npm run mock-server:build",
     "build:all": "npm run build && npm run build:auth",
     "postbuild": "dts-bundle-generator src/types/index.ts --out-file lib/index.d.ts && npm run license-checker",
     "clean": "rimraf lib dist",
-    "prepare": "npm run clean && npm run check && npm run build:all",
+    "prepublish": "npm run build:all",
```
Comment (kopijunkie) [inline package.json:23]: This script has been deprecated according to the npm docs https://docs.npmjs.com/cli/v7/using-npm/scripts#prepare-and-prepublish I think the script to run is `prepublishOnly` depending on the use case.

**Review by kopijunkie — CHANGES_REQUESTED** [review]

**Review by kopijunkie — APPROVED** [review]
LGTM 🚀

**Review by ethan605 — APPROVED** [review]
LGTM

**Pull request merged**